[self-tests] test_files.py::test_sa4 fails in `venv` environment

In ALTLinux downstream packaging stdlib's `venv` is used as an environment for running tests.

There is a test that fails in such an env.

```
Failure in test unittest.case.FunctionTestCase (runTest)                         
Traceback (most recent call last):                                               
  File "/usr/lib64/python3.12/unittest/case.py", line 58, in testPartExecutor    
    yield                                                                           
  File "/usr/lib64/python3.12/unittest/case.py", line 634, in run                
    self._callTestMethod(testMethod)                                             
  File "/usr/lib64/python3.12/unittest/case.py", line 589, in _callTestMethod    
    if method() is not None:                                                     
  File "/usr/lib64/python3.12/unittest/case.py", line 1390, in runTest           
    self._testFunc()                                                             
  File "/usr/src/RPM/BUILD/python3-module-zope.tal-5.1.1/.run_venv/lib64/python3/site-packages/zope/tal/tests/test_files.py", line 57, in runTest AssertionError("output for %s didn't match:\n%s"
AssertionError: output for /usr/src/RPM/BUILD/python3-module-zope.tal-5.1.1/.run_venv/lib64/python3/site-packages/zope/tal/tests/input/test_sa4.html didn't match:
  <!--                                                                           
  ============================================================================== 
  tests/input/test_sa4.html                                                      
  ============================================================================== 
  --><html>                                                                      
  <body>                                                                         
    <p>Some text on sa4 line 3.</p>                                              
    <!--                                                                         
  ============================================================================== 
- tests/input/test_sa3.html (line 3)                                             
+ /usr/src/RPM/BUILD/python3-module-zope.tal-5.1.1/.run_venv/lib/python3/site-packages/zope/tal/tests/input/test_sa3.html (line 3)
  ============================================================================== 
  --><div>This is macro1 on sa3 line 3.                                          
      <!--                                                                       
  ============================================================================== 
  tests/input/test_sa4.html (line 6)                                             
  ============================================================================== 
  --><b>Text from sa4 line 6 is filled into slot1.</b><!--                       
  ============================================================================== 
- tests/input/test_sa3.html (line 4)                                             
+ /usr/src/RPM/BUILD/python3-module-zope.tal-5.1.1/.run_venv/lib/python3/site-packages/zope/tal/tests/input/test_sa3.html (line 4)
  ============================================================================== 
  -->                                                                            
      This is the end of macro1 on sa3 line 5.                                   
    </div><!--                                                                   
  ============================================================================== 
  tests/input/test_sa4.html (line 8)                                             
  ============================================================================== 
  -->                                                                            
    <p>This is some text on sa4 line 9.</p>                                      
  </body>                                                                        
  </html>
```

--- a/src/zope/tal/driver.py
+++ b/src/zope/tal/driver.py
@@ -160,12 +160,12 @@ def compilefile(file, mode=None):
         else:
             mode = "xml"
     # make sure we can find the file
-    prefix = os.path.dirname(os.path.abspath(__file__)) + os.path.sep
+    prefix = os.path.dirname(os.path.realpath(__file__)) + os.path.sep
     if (not os.path.exists(file)
             and os.path.exists(os.path.join(prefix, file))):
         file = os.path.join(prefix, file)
     # normalize filenames for test output
-    filename = os.path.abspath(file)
+    filename = os.path.realpath(file)
     if filename.startswith(prefix):
         filename = filename[len(prefix):]
     filename = filename.replace(os.sep, '/')  # test files expect slashes

